test/rdcpu_sanitytest.c

```diff
@@ -16,9 +16,24 @@
 #if (defined(__i386)   || defined(__i386__)   || defined(_M_IX86) || \
      defined(__x86_64) || defined(__x86_64__) || \
      defined(_M_AMD64) || defined (_M_X64)) && defined(OPENSSL_CPUID_OBJ)
+# define IS_X_86 1
+#else
+# define IS_X_86 0
+#endif
+
+#if defined(__aarch64__)
+# define IS_AARCH_64 1
+# include "arm_arch.h"
+#else
+# define IS_AARCH_64 0
+#endif
 
+#if (IS_X_86 || IS_AARCH_64)
```

Comment:
??? Shouldn't these two cases be separate.  There is no point defining both sets of functions.

Reply:
separated the function declarations per arch. fixed.